src/model/Surface.cpp

```diff
@@ -924,8 +924,6 @@ namespace detail {
     if (test){
       test = this->setString(OS_SurfaceFields::OutsideBoundaryCondition, "OtherSideCoefficients");
       OS_ASSERT(test);
-      this->assignDefaultSunExposure();
```

Comment:
@mpraprost @kbenne  I think this one might be better fixed by adding "OtherSideCoefficients" to the list of cases checked in `assignDefaultSunExposure` and `assignDefaultWindExposure`.  This should be in the same case as if it is `Surface`.

https://github.com/NREL/OpenStudio/blob/e89a74a0178ee2e42c6c74d8f44af8f9d5510deb/src/model/Surface.cpp#L1276

Reply:
Also glad to see you getting in the C++ Marley :-)

Reply:
I think we have two potential solutions to this issue. When OtherSideCoefficients is specified, we can either:
1. not reset the sun/wind exposure settings at all (my original commit)
2. reset the sun/wind exposure to 'NoWind' and 'NoSun' (result of making the changes Dan suggested)

Do you have any input on this? Based on the original issue description, it sounded like @eringold would like to retain the original sin/wind exposure settings. @macumber do you have other insight into why solution 2 might be better?

Reply:
I actually think resetting to 'NoSun' at least would be fine. 

This came about due to perhaps another issue, which is that the Sketchup plugin allows you to set a surface outside boundary condition to 'OtherSideCoefficient' *without assigning a SurfaceProperty:OtherSideCoefficients object to that surface*, but you can't through the App or the sdk: `setOutsideBoundaryCondition` will first look for a surface property object assigned to the surface: https://github.com/NREL/OpenStudio/blob/2c4dd5c9d790d1e1a019b328ca06957660733478/src/model/Surface.cpp#L457-L459.

So I would go through my model in the plugin, set the surface boundaries to OtherSideCoefficients and make sure the sun/wind exposures are set, and then create and apply a surface property object to those surfaces with a measure, only to find that my exposures were reset!

Anyway, the EnergyPlus documentation is clear that solar effects aren't considered for OtherSideCoefficients, and like I said it will throw a warning, so at least set that to 'NoSun'. The SurfaceProperty:OtherSideCoefficients has an input for Wind Speed Coefficient, so presumably the wind effects are retained, so maybe don't reset the wind exposure, and leave that assignment up to the user.

Reply:
My suggestion was mostly just to be consistent with the pattern of how other surface types work.  It seems pretty clear that the OtherSideCoefficients option was added and we just forgot to update the cases for it.

Reply:
I agree with @macumber on this one. Can you adjust @mpraprost or let me know if you don't have the bandwidth please? 

If you do have time to adjust, please re-title the PR accordingly so it's understanble in a standalone changelog context, something like 

> Reset Surface Wind/SunExposure to NoSun/NoWind when setting Outside Boundary Condition to `OtherSideCoefficients`